docs/source/create/materials/procedural-shaders.rst

```diff
@@ -0,0 +1,387 @@
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Basic Method for Using Shaders
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Shaders are stored as a file. Fragment shaders will typically have the file extension ***.fs** or ***.frag** and vertex shaders will usually have the ***.vs** or ***.vert** file extension.
+
+Although you can associate these files with most entities, the original author of Project Athena’s end user shader implementation viewed material entities as the ideal way to apply them to other entities. As a result, this documentation will focus on using shaders with material entities.
+
+If you are unfamiliar with material entities, you can find more information `here <https://docs.projectathena.dev/create/entities/material-entity.html>`_.
+
+Material entities have data that is stored in the JSON format, and they have a property that is called ``materialData``. The ``materialData`` property will require model and procedural fields. Here is an example::
+
+    {
+        "materials": [{
+            "model": "hifi_shader_simple",
+            "procedural": {
+              "version": 3,
+              "fragmentShaderURL": "https://gist.githubusercontent.com/MarcusLlewellyn/2726885d4cc8be086a6755f05fbd99ff/raw/rotatecube.fs"
+            }
+        }]
+    }
+
+The ``materialData`` JSON can be applied either via the Project Athena Interface’s edit tools or with a script. Here's another example::
+
+    Entities.addEntity({
+    	type: "Material",
+    	materialURL: "materialData",
+    	priority: 1,
+    	materialData: JSON.stringify({
+    		materials: {
+    			"model": "hifi_shader_simple"
+    			"procedural": {
+    			  	"version": 3,
+    			  	"shaderUrl": "https://gist.githubusercontent.com/SamGondelman/8bbd39f91d20cab4c75280d9b1cb0764/raw/7930289654ce8309bbe785907f03eabc1dbc6181/Proceduralv3.fs"
+    			}
+    		}
+    	})
+    });
+    
+You must specify the material "model" as ``hifi_shader_simple`` and provide a shader link. To provide a fragment shader, set ``fragmentShaderURL`` (or ``shaderUrl``). To provide a vertex shader, set ``vertexShaderURL``.
+
+^^^^^^^^^^^^^^^
+Shader Template
+^^^^^^^^^^^^^^^
+
+When you learn about shaders for other applications, the shader may have a function like ``main()`` that is run first. By contrast, Project Athena has a specific function name that must be called. Which function is used depends on which version of the shader you use.
+
+As shaders were developed, features for them evolved a bit over time. As a result, there are several shader versions, and each version has a different call signature. **Versions 1 and 2** are the oldest, and will still work. **Versions 3 and 4** are the newest and expose more features. Version 4 provides for per-fragment positions, however it is also the most expensive. Therefore it is recommended to use Version 3 if that extra feature from Version 4 is not needed.
+
+The most basic template for a shader will look something like this example::
```

Comment:
this example should be a v3 shader

Reply:
also two : here

Reply:
and elsewhere

Reply:
I will need a very simple example that can be used as a starting point for a v3 shader to replace that. :)